zellij-tile/src/data.rs

```diff
@@ -114,6 +117,7 @@ pub struct ModeInfo {
     // FIXME: This should probably return Keys and Actions, then sort out strings plugin-side
     pub keybinds: Vec<(String, String)>, // <shortcut> => <shortcut description>
     pub palette: Palette,
+    pub input_prompt: String, // input_prompt is used by modes that require input displays, e.g. tab search or rename
```

Comment:
Similar to above here, this feels very specific to this change. I think this would be better done in the tab-bar itself. The tab-bar just needs to check for the `SearchTab` mode and then it is in charge of the prompt. I don't really think that the prompt should be something Zellij decides anyways.

What do you think?

Reply:
I generally agree with this and will probably revisit it this weekend. Everything I did on the display side of things felt pretty hacky to me. I do think input prompts could be reasonably tied to modes IF there will be future features that depend on text input but I don't know what the best place to put it would be.

Reply:
This and your further comment below make me think I need to take a closer look at this soon! I may not be understanding the actually UI here! I'll hopefully get back to you somewhat soon!

Reply:
Whoops! Turns out my "somewhat soon" was right now! I'll admit I feel a little uneasy about having the prompt appear in the status bar when things are tab related – especially given people can run zellij with a tab-bar but without a status-bar. It would mean some rendering changes in the tab bar, but it does feel like it belongs there more than the status bar, but I don't know if that is something you and @a-kenji already discussed!

Reply:
> people can run zellij with a tab-bar but without a status-bar

Good to know! I actually wasn't aware of this.

Let me take a look tonight and see if I can come up with a tab bar rendering option. If I recall correctly, the tab bar scrolls if there's not enough space to display them all so it could just be a matter of reserving some space at the end of the bar. I'll need to take a closer look to be sure.